### ISSUE TITLE
Joomgallery Menu Items should take Precedence of URL

### ISSUE THREAD

**Issue by JoshuaLewis** [JoomGallery/JoomGallery#27]
When I create a menu item for a specific category, the url of the menu item does not take precedence. With native Joomla articles if one creates a menu item for that article, the category view will literally display a url of site.com/article-name. As a result Joomla allows for one to create simple url's via menu items and the categories themselves use the menu item ones instead of site.com/category/article-name.

As for a real life example of how Joomgallery displays the url, you can see my "Favorite" album not using the menu item created for the Favorite album: http://www.joshlewis.us/photo

The menu item of course works as expected as seen here: http://www.joshlewis.us/favorite
The issue is that the main gallery menu item does not want to use the menu item url for specific categories, as a result no one will actually see the short url that I'm trying to produce due to the main gallery displaying the old ones.

**Comment by Chraneco** [comment]
Yes, that is a known issue. Unfortunately, fixing this requires changes at a lot of places. According to a recent news article Joomla! 3.5 will contain a new routing system:
http://developer.joomla.org/news/614-joomla-3-5-update.html

I think it's best to wait for that and overhaul the whole routing in JoomGallery all at once. This will hopefully fix also a number of other problems.

**Comment by JoshuaLewis** [comment]
Awesome, was waiting a long time to hear about the new routing system. I should send a reminder about this after it's implemented.
